deploy/src/Microsoft.Azure.IIoT.Deployment/Authentication/IAuthenticationManager.cs

```diff
@@ -0,0 +1,41 @@
+﻿// ------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All rights reserved.
+//  Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+namespace Microsoft.Azure.IIoT.Deployment.Authentication {
+
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    using Microsoft.Azure.Management.ResourceManager.Fluent;
+    using Microsoft.Identity.Client;
+
+    public interface IAuthenticationManager {
+
+        public  Task AuthenticateAsync(
```

Comment:
Would be great to get the public interfaces doc'ed to some degree, at least for it does and how it is supposed to be used.

Reply:
Done.